core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -330,19 +332,21 @@ object SparkSubmit extends CommandLineUtils {
     args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
 
     // In client mode, download remote files.
+    var localPrimaryResource: String = null
+    var localJars: String = null
+    var localPyFiles: String = null
```

Comment:
I'm curious why we don't download files and archives here as well?  I guess for archives we would want to extract as well to keep same behavior at least for yarn. I guess this was originally for standalone and mesos mode but it seems like for consistency we should download archives.  

It looks like the original one did download files? Did that get intentionally dropped?
https://github.com/apache/spark/commit/4af37812915763ac3bfd91a600a7f00a4b84d29a